src/geo/spatial.clj

```diff
@@ -276,24 +278,28 @@
          degrees (radians->degrees radians)]
      (GeoCircle. point degrees earth))))
 
+(defn distance-in-degrees
+  "Distance between two points, in degrees."
+  [a b]
+  (-> vincenty-distance-calculator
+      (.distance (to-spatial4j-point a)
+                 (to-spatial4j-point b))))
+
 (defn distance
   "Distance between two points, in meters"
   [a b]
   ; There's a singularity in the geohash library's distance calculation
-  ; algorithm, used here, which causes distances near the poles to return NaN.
-  (assert (not= 90.0 (abs (latitude a))))
-  (assert (not= 90.0 (abs (latitude b))))
-  (VincentyGeodesy/distanceInMeters
-   (to-geohash-point a)
-   (to-geohash-point b)))
-
-(defn distance-in-degrees
-  "Distance between two points, in degrees."
-  [a b]
-  (-> earth
-    .getDistCalc
-    (.distance (to-spatial4j-point a)
-               (to-spatial4j-point b))))
+  ; algorithm, which causes distances at the poles to return
+  ; bad results. In these cases, use spatial4j's spherical vincenty
+  ; distance calculator.
+ (if (and (not= (double (abs (latitude a))) 90.0)
```

Comment:
I'm afraid I'm not informed enough about the precise problem here to know the answer to this, but it occurs to me to wonder whether this is truly an "exactly 90.0" case or if there is any need to check for other points within a small enough delta. I see we have the existing test case that compared `89.99999` vs `90.0`, so I guess that suggests it is OK as is. Do you have any insight into that?

Reply:
That's a good catch -- I just pushed another commit. On my machine at least, going higher than 89.99... with 14 decimal points is where geohash starts to overflow. Setting that as the threshold should be a good point to switch to spatial4j.